### DISCUSSION TITLE
Cant grab query params through `req.url.searchParams` anymore in the latest version

### DISCUSSION THREAD

**Discussion by Hud97** [mswjs/msw#1358] [Q&A] [answered]
Hi folks 😊

After I upgraded msw to latest version, I cannot invoke `req.url.searchParams` in the callback resolvers to grab query params and it doesnt mention else where in the doc for the latest version. any workaround ?

**Comment by kettanaito** [comment]
Hey, @Hud97. 

There haven't been any changes around that, so I'd expect it to work. Do you have a reproduction repository/sandbox where I can look at your issue?

**Reply by Hud97** [reply]
Unfortunately I dont currently, its in my company codebase and its private. Its probably working but types does not provide autocompletion for `req.url` and I assumed its not working, my bad 😅.

**Reply by kettanaito** [reply]
Yeah, you cannot annotate URL search parameters because you don't need to—those are always either a string or an array of strings (when using `.getAll()`), or `null` if the referenced parameter does not exist.

**Comment by kettanaito** [comment, marked as answer]
The issue has been resolved, it was confusion about the lack of types for `req.url`. URL search parameters cannot be explicitly annotated and that's a good thing.
